python/cudf/cudf/core/window/rolling.py

```diff
@@ -192,9 +193,26 @@ def __getitem__(self, arg):
         )
 
     def _apply_agg_series(self, sr, agg_name):
-        result_col = libcudf.rolling.rolling(
-            sr._column, self.window, self.min_periods, self.center, agg_name
-        )
+        if isinstance(self.window, int):
+            result_col = libcudf.rolling.rolling(
+                sr._column,
+                None,
+                None,
+                self.window,
+                self.min_periods,
+                self.center,
+                agg_name,
+            )
+        else:
+            result_col = libcudf.rolling.rolling(
+                sr._column,
+                as_column(self.window),
+                as_column([0] * self.window.size, dtype=self.window.dtype),
```

Comment:
This generates a Python list of size `self.window.size` to construct the column which will be really slow if `self.window.size` is large. I think we have a column utility to generate a column of size N on the GPU.